apps/site/src/lib/api/model/api-key.model.ts

```diff
@@ -206,13 +211,42 @@ export class ApiKey {
     return apiKey;
   }
 
-  static async revokeAll(db: Db, params: { user: User }): Promise<void> {
-    await db
+  static async updateByUser(
+    db: Db,
+    params: {
+      publicId: string;
+      user: User;
+      displayName?: string;
+      revokedAt?: Date;
+    },
+  ): Promise<boolean> {
+    const response = await db
       .collection<ApiKeyDocument>(ApiKey.COLLECTION_NAME)
-      .updateMany(
-        { user: params.user.toRef(), revokedAt: { $eq: null } },
-        { $set: { revokedAt: new Date() } },
+      .updateOne(
+        {
+          user: params.user.toRef(),
+          publicId: params.publicId,
+          revokedAt: { $eq: null },
```

Comment:
This means that users cannot rename a key once it has been revoked – I'm not sure we need to impose that restriction (users might want to revoke an old 'Production' key and then rename it to 'Production – old' or something). I agree that we don't want to revoke keys that have already been revoked because it makes the timestamp misleading.

Reply:
I see, I thought we didn't return the revoked keys in the `List` endpoint, but that's just because we filter them away in the UI. I'll adjust this

Reply:
I've fixed this in 1d0b2c7713dd297c38d9d106f04dd275ae670db6
also now treat idempotent updates as successful